DOCSP-43087: Atlas Search Indexes (#58)
mongodb/docs-c: source/indexes/atlas-search-index.txt

.. _c-atlas-search-index:

====================
Atlas Search Indexes
====================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

.. facet::
   :name: genre
   :values: reference

.. meta::
   :keywords: index, query, optimization, efficiency

Overview
--------

:atlas:`Atlas Search </atlas-search>` enables you to perform full-text searches on
collections hosted on MongoDB Atlas. Atlas Search indexes specify the behavior of
the search and which fields to index.

The following sections provide code examples that demonstrate how to create, list, update,
and delete Atlas Search indexes.

.. _c-atlas-search-index-create:

Create a Search Index
---------------------

> You can use the ``mongoc_collection_command_simple()`` function to create one or more
> Atlas Search indexes by using the ``createSearchIndexes`` command.

Reviewer: S: to avoid the use/using repetition:

```suggestion
You can pass the ``createSearchIndexes`` command to the ``mongoc_collection_command_simple()`` function to create one or more
Atlas Search indexes.
```

> You can also use this function to create Atlas Vector Search Indexes.

Reviewer: S:
```suggestion
You can also use this function to create Atlas Vector Search indexes.
```

Atlas Vector Search enables you to perform semantic searches on vector
embeddings stored in MongoDB Atlas. To learn more about this feature,
see the :atlas:`Atlas Vector Search Overview

> </atlas-vector-search/vector-search-overview/>`.

Reviewer: S: clarify where this link is going
```suggestion
</atlas-vector-search/vector-search-overview/>` in the Atlas documentation.
```

The following code example shows how to create an Atlas Search index:

.. literalinclude:: /includes/indexes/indexes.c
   :language: c
   :start-after: start-create-search-index
   :end-before: end-create-search-index
   :dedent:

The following code example shows how to create multiple indexes:

.. literalinclude:: /includes/indexes/indexes.c
   :language: c
   :start-after: start-create-search-indexes
   :end-before: end-create-search-indexes
   :dedent:

To learn more about the syntax used to define Atlas Search indexes, see the
:atlas:`Review Atlas Search Index Syntax </atlas-search/index-definitions>` guide

> in the Atlas manual.

Reviewer: S: I think we usually just say Atlas documentation (and Server manual)

```suggestion
in the Atlas documentation.
```

.. _c-atlas-search-index-list:

List Search Indexes
-------------------

> You can use the
> ``mongoc_collection_aggregate()`` function to return all Atlas Search indexes in a
> collection by using the ``$listSearchIndexes`` aggregation stage.

Reviewer: S: similar suggestion as above
```suggestion
You can pass the ``$listSearchIndexes`` aggregation stage to the
``mongoc_collection_aggregate()`` function to return all Atlas Search indexes in a
collection.
```

Reviewer: Applies to the intro paragraph of the next two sections as well
